third_party/nvfuser/csrc/python_frontend/fusion_cache.h

```diff
@@ -11,6 +11,32 @@ namespace Nvf = torch::jit::fuser::cuda;
 
 namespace nvfuser {
 
+//! \struct UserDefinedFusionSchedule
+//! \brief A container to hold a scheduled Fusion IR as well as an executor
+//! to contain the corresponding geerated kernel.
+struct UserDefinedFusionSchedule {
+  //! Scheduled Fusion IR
+  std::unique_ptr<Nvf::Fusion> schedule;
+  //! Generated kernel container
+  std::unique_ptr<Nvf::FusionExecutor> executor;
+};
+
+//! \struct FusionSchedules
+//! \brief A container for auto generated and user defined schedules
+//! that correspond to compiled kernels for each complete Fusion Definition.
+struct FusionSchedules {
+  FusionSchedules();
+  Nvf::Fusion* preschedFusion();
+
+  // Schedules Automatically generated by nvFuser for dynamic inputs. (default)
+  // NOTE: The FusionExecutorCache also holds the Unscheduled Fusion IR
+  std::unique_ptr<Nvf::FusionExecutorCache> auto_gen_schedules;
+  // Schedules defined by the user for specific input sizes.
+  // They are also generated per device as all devices may not be the same.
+  std::unordered_map<size_t, std::vector<UserDefinedFusionSchedule>>
```

Comment:
comment on the structure here is a little vague. Can we make it more explicit on what the key is and what does the index mean for std::vector?

Reply:
similarly, `user_defined_schedules` is not used at all neither.

Reply:
Comment added.  

`user_defined_schedules` are not used, yet.